controllers/os-common/pkg/actuator/actuator.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actuator
+
+import (
+	"context"
+
+	"github.com/gardener/gardener-extensions/controllers/os-common/pkg/generator"
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type actuator struct {
+	scheme    *runtime.Scheme
+	client    client.Client
+	logger    logr.Logger
+	osName    string
+	generator generator.Generator
+}
+
+// NewActuator creates a new actuator with the given logger.
+func NewActuator(osName string, generator generator.Generator) operatingsystemconfig.Actuator {
+	return &actuator{
+		logger:    log.Log.WithName(osName + "-operatingsystemconfig-actuator"),
+		osName:    osName,
+		generator: generator,
+	}
+}
+
+func (a *actuator) InjectScheme(scheme *runtime.Scheme) error {
```

Comment:
is `scheme` used for anything?

Reply:
Most of this code is copied from Alicloud's controller. I think @adracus can comment on this better.